### ISSUE TITLE
[BUG] ImagePredictor is too slow with fit

### ISSUE THREAD

**Issue by serhiyandrejev** [autogluon/autogluon#2110]
**Describe the bug**
Im using RTX 3060. CUDA is ok, torch is ok. But when I fit the model it looks like CPU fitting. Then, my DataSpell just turn off. 

>>> import torch
>>> torch.cuda.is_available()
True
>>> torch.cuda.get_device_name(0)
'NVIDIA GeForce RTX 3060'

**Screenshots**
![image](https://user-images.githubusercontent.com/67690614/188890065-24ab4f22-3fd2-4d2b-9447-b63a82119ffb.png)

**Installed Versions**
<details>
INSTALLED VERSIONS
------------------
date                   : 2022-09-07
time                   : 16:29:55.627892
python                 : 3.9.13.final.0
OS                     : Windows
OS-release             : 10
Version                : 10.0.19044
machine                : AMD64
processor              : AMD64 Family 25 Model 33 Stepping 0, AuthenticAMD
num_cores              : 16
cpu_ram_mb             : 49096
cuda version           : None
num_gpus               : 1
gpu_ram_mb             : [9043]
avail_disk_size_mb     : None

autogluon.common       : 0.5.2
autogluon.core         : 0.5.2
autogluon.features     : 0.5.2
autogluon.multimodal   : 0.5.2
autogluon.tabular      : 0.5.2
autogluon.text         : 0.5.2
autogluon.timeseries   : None
autogluon.vision       : 0.5.2
autogluon_contrib_nlp  : None
boto3                  : 1.24.65
catboost               : 1.0.6
dask                   : 2021.11.2
distributed            : 2021.11.2
fairscale              : 0.4.6
fastai                 : 2.7.9
gluoncv                : 0.11.0
hyperopt               : 0.2.7
lightgbm               : 3.3.2
matplotlib             : 3.5.3
networkx               : 2.8.6
nlpaug                 : 1.1.10
nltk                   : 3.7
nptyping               : 1.4.4
numpy                  : 1.21.6
omegaconf              : 2.1.2
pandas                 : 1.4.4
PIL                    : 9.0.1
protobuf               : None
psutil                 : 5.8.0
pytorch-metric-learning: None
pytorch_lightning      : 1.6.5
ray                    : 1.13.0
requests               : 2.28.1
scipy                  : 1.7.3
sentencepiece          : None
skimage                : 0.19.3
sklearn                : 1.0.2
smart_open             : 5.2.1
timm                   : 0.5.4
torch                  : 1.12.0+cu113
torchmetrics           : 0.7.3
torchtext              : 0.13.0
torchvision            : 0.13.0+cu113
tqdm                   : 4.64.0
transformers           : 4.20.1
xgboost                : 1.4.2
</details>

**Comment by ThatDRW** [comment]
It's hard to tell exactly what's going on for me, but I hope this might help you.

If I'm not mistaken, Torch defaults tensors and models to be generated on CPU, even if `torch.cuda.is_available()` returns `True`.

I found the following on [wandb](https://wandb.ai/wandb/common-ml-errors/reports/How-To-Use-GPU-with-PyTorch---VmlldzozMzAxMDk)
![afbeelding](https://user-images.githubusercontent.com/112971591/188896879-22b6e912-007a-46c2-aa17-ebf106dad3bc.png)

I suggest checking if the data and model are in fact initialized on the GPU with `.is_cuda` and if not, using the `.to(device)` to move it to your GPU.

**Comment by Innixma** [comment]
@serhiyandrejev we are currently migrating all vision logic to autogluon.multimodal . Can you try fitting your dataset with MultimodalPredictor? It might be that vision module doesn't pickup GPU properly on Windows.

**Comment by bryanyzhu** [comment]
Another thing is, it looks like you are doing HPO too. [`good_quality_fast_inference`](https://github.com/awslabs/autogluon/blob/master/vision/src/autogluon/vision/predictor/predictor.py#L137) contains three models and multiple parameters to search. They are all efficient models for fast inference, but it doesn't mean training is fast. In order to debug, I suggest to just train on one model and see if the speed makes sense.

**Comment by Innixma** [comment]
Marking as resolved, vision logic has been moved to autogluon.multimodal
